src/SignalR/clients/ts/signalr/tests/HubConnection.test.ts

```diff
@@ -1075,6 +1075,8 @@ describe("HubConnection", () => {
                     // Observer should no longer receive messages
                     expect(observer.itemsReceived).toEqual([1]);
 
+                    // Close message sent asynchronously so we need to wait
+                    await delay(50);
```

Comment:
Do you want flaky tests? Because this is how you get flaky tests.

You need a much higher timeout (like 30 seconds) and a way to complete early if the close message arrives before the timeout.

Reply:
I can do the setInterval pattern
```
setInterval(() => {
    //check value we're waiting for and dispose interval if found or took too long
}, 10);
```